lib/parser.rb

```diff
@@ -18,14 +18,19 @@ def advance
     @command = @lines.shift
   end
 
-  NON_ARITHMETIC_COMMANDS = ["push", "pop", "label", "if-goto", "goto"]
+  NON_ARITHMETIC_COMMAND_TYPES = {
+    'push' => :C_PUSH,
+    'pop' => :C_POP,
+    'label' => :C_LABEL,
+    'if-goto' => :C_IF,
+    'goto' => :C_GOTO
+  }
 
   def command_type
     raw_command = @command.split[0]
-
     case raw_command
-    when *NON_ARITHMETIC_COMMANDS
-      "C_#{raw_command.split("-")[0].upcase}".to_sym
+    when *NON_ARITHMETIC_COMMAND_TYPES.keys
+      NON_ARITHMETIC_COMMAND_TYPES[raw_command]
```

Comment:
Nice, this is clearer I think. WDYT about the `Hash#fetch` trick I [mentioned](https://github.com/wildmaples/jack-virtual-machine/pull/14#discussion_r718264168)?